### PR TITLE
fix: incorrect currency displayed in profiledropdown

### LINKED ISSUE #4056: Correct currency displayed in profiledropdown
On Beta this happen to me:
**BSX**
Screenshot 2022-09-30 at 08-35-06 KodaDot - Kusama NFT Market Explorer

**SNEK**
Screenshot 2022-09-30 at 08-30-14 Fancy cat KodaDot - Kusama NFT Market Explorer

Is it normal we KSM instead of BSX?

_Originally posted by @roiLeo in https://github.com/kodadot/nft-gallery/issues/3395#issuecomment-1263166887_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #4056 
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
BSX:

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
image

image

image

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
wfm
Capture d’écran 2022-10-19 à 8 32 25 AM

could you check why tests are failing?

File: components/rmrk/Gallery/CollectionDetail.vue [hunk 1]
```diff
@@ -30,15 +30,29 @@
...
+              v-if="tokenId"
+              :value="collectionFloorPrice"
+              :token-id="tokenId"
+              :prefix="urlPrefix"
+              inline />
             <Money
+              v-else
               :value="collectionFloorPrice"
               inline
               data-cy="collection-floor-price" />
```
Comment (vikiival) [inline components/rmrk/Gallery/CollectionDetail.vue:58]: Unify under one component (check AccountBalance.vue)

**Review by vikiival — CHANGES_REQUESTED** [review]
PLease DRY

File: utils/mixins/prefixMixin.ts [hunk 2]
```diff
@@ -18,4 +19,8 @@ export default class PrefixMixin extends Vue {
   get isMoonriver(): boolean {
     return this.urlPrefix === 'movr'
   }
+
+  get tokenId() {
+    return getKusamaAssetId(this.urlPrefix)
```
Comment (vikiival) [inline utils/mixins/prefixMixin.ts:24]: `getKusamaAssetId` should handle also other chains

File: components/rmrk/Gallery/CollectionDetail.vue (hunk 1 above)
Comment (Jarsen136) [reply components/rmrk/Gallery/CollectionDetail.vue]: Ok, I made a CommonTokenMoney Component.

File: utils/mixins/prefixMixin.ts (hunk 2 above)
Comment (Jarsen136) [reply utils/mixins/prefixMixin.ts:24]: For now, the `CommonTokenMoney ` could handle both `rmrk\bsx\snek` chains. What else chain should I add? 👀

Comment (yangwao) [reply utils/mixins/prefixMixin.ts:24]: in the future there will be movr/glmr and astar

File: components/shared/CommonTokenMoney.vue [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <TokenMoney
+    v-if="tokenId"
```
Comment (vikiival) [inline components/shared/CommonTokenMoney.vue:3]: tokenId isnt prop?

**Review by vikiival — CHANGES_REQUESTED** [review]

File: components/shared/CommonTokenMoney.vue (hunk 3 above)
Comment (Jarsen136) [reply components/shared/CommonTokenMoney.vue:3]: tokenId is from `PrefixMixin`  rather than the props

File: utils/mixins/prefixMixin.ts (hunk 2 above)
Comment (Jarsen136) [reply utils/mixins/prefixMixin.ts:24]: Yup, but I'm not sure which token should be used in these chains.

I guess the asset list query should also add more tokens in the future. Or if I have miss something?
image

Comment (vikiival) [reply utils/mixins/prefixMixin.ts:24]: return zero for now please

Comment (Jarsen136) [reply utils/mixins/prefixMixin.ts:24]: OK, I will check.

**Comment by Jarsen136** [comment]
> hmm, I guess it's not possible to have bsx on rmrk :)
> 

Good catch! Gallery items from different chains may display on the `Newest list` at the same time.

✅ FIXED
image

File: components/shared/CommonTokenMoney.vue (hunk 3 above)
Comment (Jarsen136) [reply components/shared/CommonTokenMoney.vue:3]: tokenId is props now

**Comment by yangwao** [comment]
1.0000 KSM ? Like should be 1 KSM I guess?

image

**Comment by Jarsen136** [comment]
> 1.0000 KSM ? Like should be 1 KSM I guess?


on bsx:

image

on rmrk:

image


It seem related to https://github.com/kodadot/nft-gallery/issues/3868

**Comment by yangwao** [comment]
hmm so till then I guess we can merge it?

**Comment by yangwao** [comment]
let's see 

pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 34.44 USD/KSM ~ 1.452 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x0cddd54ad5108d54817ded1dd3fa96ce06ed92713e5b3b45f65b223aac85221e](https://kusama.subscan.io/extrinsic/0x0cddd54ad5108d54817ded1dd3fa96ce06ed92713e5b3b45f65b223aac85221e)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**